libraries/AP_Scripting/examples/Aerobatics/Trajectory/plane_aerobatics.lua

```diff
@@ -1285,6 +1338,147 @@ function create_auto_mission()
    end
 end
 
+function PathTask(fn, name, id, initial_yaw_deg, arg1, arg2, arg3, arg4)
+   self = {}
+   self.fn = fn
+   self.name = name
+   self.id = id
+   self.initial_yaw_deg = initial_yaw_deg
+   self.args = { arg1, arg2, arg3, arg4 }
+   self.started = false
+   return self
+end
+
+-- see if an auto mission item needs to be run
+function check_auto_mission()
+   id, cmd, arg1, arg2, arg3, arg4 = vehicle:nav_script_time()
+   if not id then
+      return
+   end
+   if id ~= last_id then
+      -- we've started a new command
+      current_task = nil
+      last_id = id
+      initial_yaw_deg = math.deg(ahrs:get_yaw())
+      gcs:send_text(0, string.format("Starting %s!", command_table[cmd].name ))
+
+      initial_height = ahrs:get_position():alt()*0.01
+      -- work out yaw between previous WP and next WP
+      local cnum = mission:get_current_nav_index()
+      -- find previous nav waypoint
+      local loc_prev = get_wp_location(cnum-1)
+      local loc_next = get_wp_location(cnum+1)
+      local i= cnum-1
+      while get_wp_location(i):lat() == 0 and get_wp_location(i):lng() == 0 do
+         i = i-1
+         loc_prev = get_wp_location(i)
+      end
+      -- find next nav waypoint
+      i = cnum+1
+      while get_wp_location(i):lat() == 0 and get_wp_location(i):lng() == 0 do
+         i = i+1
+         loc_next = get_wp_location(resolve_jump(i))
+      end
+      wp_yaw_deg = math.deg(loc_prev:get_bearing(loc_next))
+      if math.abs(wrap_180(initial_yaw_deg - wp_yaw_deg)) > 90 then
+         gcs:send_text(0, string.format("Doing turnaround!"))
+         wp_yaw_deg = wrap_180(wp_yaw_deg + 180)
+      end
+      initial_yaw_deg = wp_yaw_deg
+      current_task = PathTask(command_table[cmd].fn, command_table[cmd].name,
+                              id, initial_yaw_deg, arg1, arg2, arg3, arg4)
+   end
+end
+
+local last_trick_action_state = 0
+local trick_sel_chan = nil
+local last_trick_selection = nil
+
+--[[
+   get selected trick. Trick numbers are 1 .. TRIK_COUNT. A value of 0 is invalid
+--]]
+function get_trick_selection()
+   if trick_sel_chan == nil then
+      trick_sel_chan = rc:find_channel_for_option(TRIK_SEL_FN:get())
+      if trick_sel_chan == nil then
+         return 0
+      end
+   end
+   -- get trick selection based on selection channel input and number of tricks
+   local i = math.floor(TRIK_COUNT:get() * 0.5*(trick_sel_chan:norm_input_ignore_trim()+1))+1
```

Comment:
I think this needs to be changed...wont work with a normal three position switch or pot if the high position is even slightly less than RCx_MAX....bottom is okay...

Reply:
found this...dont know if its in our LUA
```suggestion
   local i = math.floor(TRIK_COUNT:get() * 0.5*(trick_sel_chan:norm_input_ignore_trim()+1))+1
   i = math.clamp(i, 1, TRIK_COUNT:get())
```

Reply:
also cruise alt hold is messed up I think...it creeps....

Reply:
thanks, fixed and tested, going beyond min/max now works